src/helpers/update-tasks.ts

```diff
@@ -1,111 +1,153 @@
-  } else {
-    logger.info(`Passed the deadline on ${issue.url} and no activity is detected, removing assignees.`);
-    if (await removeAllAssignees(context, issue)) {
-      await supabase.issues.delete(issue.url);
-    }
+  const { projectUrls, repos } = await getProjectUrls(context, watch);
+
+  if (!projectUrls?.length && !repos?.length) {
+    logger.info("No watched repos have been found, no work to do.");
+    return false;
+  }
+
+  for (const repo of repos) {
+    logger.info(`Updating reminders for ${repo.owner.login}/${repo.name}`);
+    await updateReminders(context, repo);
   }
+
+  return true;
 }
 
-async function remindAssigneesForIssue(context: Context, issue: Database["public"]["Tables"]["issues"]["Row"]) {
+async function updateReminders(context: Context, repo: ListForOrg["data"][0]) {
   const {
-    adapters: { supabase },
-    logger,
-    config,
+    octokit
   } = context;
-  const now = DateTime.now();
-  const deadline = DateTime.fromISO(issue.deadline);
+  const issues = await octokit.paginate(octokit.rest.issues.listForRepo, {
+    owner: repo.owner.login,
+    repo: repo.name,
+    per_page: 100,
+    state: "open",
+  }) as ListIssueForRepo[];
 
-  if (config.warning <= 0) {
-    logger.info("The reminder threshold is <= 0, won't send any reminder.");
-  } else {
-    const lastReminder = issue.last_reminder;
-    logger.info(`We are passed the deadline on ${issue.url}, should we send a reminder? ${!lastReminder}`);
-    if (!lastReminder && (await remindAssignees(context, issue))) {
-      await supabase.issues.upsert({
-        url: issue.url,
-        deadline: deadline.toJSDate(),
-        lastReminder: now.toJSDate(),
-        lastCheck: now.toJSDate(),
-      });
+  for (const issue of issues) {
+    if (issue.assignees?.length || issue.assignee) {
+      await updateReminderForIssue(context, repo, issue);
     }
   }
 }
 
-async function updateReminders(context: Context, issue: Database["public"]["Tables"]["issues"]["Row"]) {
+async function updateReminderForIssue(context: Context, repo: ListForOrg["data"][0], issue: ListIssueForRepo) {
   const {
-    adapters: { supabase },
     logger,
     config,
     payload,
+    octokit
   } = context;
+  const comments = await octokit.paginate(octokit.rest.issues.listComments, {
+    owner: repo.owner.login,
+    repo: repo.name,
+    issue_number: issue.number,
+    per_page: 100,
+  }) as ListCommentsForIssue[];
+
+  const botComments = comments.filter((o) => o.user?.type === "Bot");
+  const dateRegex = /(?<=<td>)(\w{3}, \w{3} \d{1,2}, \d{1,2}:\d{2} (AM|PM) UTC)(?=<\/td>)/gi;
+  const assignmentRegex = /Ubiquity - Assignment - start -/gi;
+  const botAssignmentComments = sortAndReturn(botComments.filter((o) => assignmentRegex.test(o?.body || "")), "desc");
+  const botFollowup = /this task has been idle for a while. Please provide an update./gi;
+  const botFollowupComments = botComments.filter((o) => botFollowup.test(o?.body || ""));
```

Comment:
I had a generalized function for writing metadata that would add a "comment header" which included the function caller's name. This was intended to be used as a target for future metadata.

Reply:
That is what you are seeing here is it not?

```ts
const assignmentRegex = /Ubiquity - Assignment - start -/gi;
```

```txt
<!-- Ubiquity - Assignment - start - 86f94c4
{
  "duration": 14400,
  "priceLabel": {
    "id": 7036610424,
    "node_id": "LA_kwDOL-In888AAAABo2oneA",
    "url": "https://api.github.com/repos/ubiquibot/command-start-stop/labels/Price:%20200%20USD",
    "name": "Price: 200 USD",
    "color": "1f883d",
    "default": false,
    "description": null
  }
}
-->
```

https://github.com/ubiquibot/command-start-stop/issues/2#issuecomment-2227499243

>  It'd be better if we store data in the comment metadata but I'm not sure if we have this functionality implemented.

I forgot to last night but I will add that minor change into one of my open PRs and include the deadline, the userID etc and it can be mapped easier

Reply:
I see, this metadata is generated by another plugin, right? Ideally it should be structured like JSON or something similar.
However my point still stands for bot follow-up regex but this can be fixed in a later PR

Reply:
I just saw this thread, I had the same concern myself
https://github.com/ubiquibot/user-activity-watcher/pull/13#discussion_r1721376180

Even by using metadata checking for a string seems unreliable and easily breakable. What is the purpose of checking for this bot comment?